projectorganizer/src/prjorg-project.c

```diff
@@ -497,13 +500,49 @@ gchar **prjorg_project_load_expanded_paths(GKeyFile * key_file)
 }
 
 
+static GPtrArray *get_session_files(GKeyFile *config)
+{
+	guint i;
+	gboolean have_session_files;
+	gchar entry[16];
+	gchar **tmp_array;
+	GError *error = NULL;
+	GPtrArray *files;
+	gchar *unescaped_filename;
+
+	files = g_ptr_array_new();
+	have_session_files = TRUE;
+	i = 0;
+	while (have_session_files)
+	{
+		g_snprintf(entry, sizeof(entry), "FILE_NAME_%d", i);
+		tmp_array = g_key_file_get_string_list(config, "files", entry, NULL, &error);
+		if (! tmp_array || error)
+		{
+			g_error_free(error);
+			error = NULL;
+			have_session_files = FALSE;
+		} else {
+			unescaped_filename = g_uri_unescape_string(tmp_array[7], NULL);
+			g_ptr_array_add(files, g_strdup(unescaped_filename));
+			g_free(unescaped_filename);
+		}
+		i++;
+	}
+	g_ptr_array_add(files, NULL);
+
+	return files;
```

Comment:
change to `return g_ptr_array_free(files, FALSE);` and return `gchar **` from the function.